javaslang-test/src/main/java/javaslang/test/Gen.java

```diff
@@ -277,6 +310,17 @@
         return choose(1, size).flatMap(n -> new Frequency().gen(n, stream));
     }
 
+  /**
+   * Intersperse values from this generator instance with those of another.
+   *
+   * @param other another T generator to accept values from.
+   * @return A new T generator
+   */
+    default Gen<T> intersperse(Gen<T> other) {
+        AtomicReference<Tuple2<Gen<T>, Gen<T>>> interspersedReference = new AtomicReference<>(Tuple.of(other, this));
```

Comment:
Using an `AtomicReference` works, but I feel somewhat dirty using it. But due to immutability and `final` references from inside lambdas, I wanted something I could update an order swapped tuple.